File: BankingApp_WDP.API/DTOs/Transactions/TransactionResponse.cs
using BankingApp.Models;

namespace BankingApp.DTOs.Transactions;

public class TransactionResponse
{
    public int Id { get; set; }
    public int? FromAccountId { get; set; }
    public int? ToAccountId { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ProcessedAt { get; set; }

    public static TransactionResponse FromTransaction(Transaction tx) => new()
    {
        Id = tx.Id,
        FromAccountId = tx.FromAccountId,
        ToAccountId = tx.ToAccountId,
        Amount = tx.Amount,
        Currency = tx.Currency,
        Type = tx.Type.ToString(),
        Status = tx.Status.ToString(),
        Description = tx.Description,
        CreatedAt = tx.CreatedAt,
        ProcessedAt = tx.ProcessedAt
    };
}

File: BankingApp_WDP.APP/app/src/main/java/com/example/bankingapp/data/model/transaction/TransactionResponse.kt
package com.example.bankingapp.data.model.transaction

import com.google.gson.annotations.SerializedName

data class TransactionResponse(
    @SerializedName("id")            val id: String,
    @SerializedName("fromAccountId") val fromAccountId: String?,
    @SerializedName("toAccountId")   val toAccountId: String?,
    @SerializedName("amount")        val amount: Double,
    @SerializedName("currency")      val currency: String,
    @SerializedName("type")          val type: String,
    @SerializedName("status")        val status: String,
    @SerializedName("description")   val description: String?,
    @SerializedName("createdAt")     val createdAt: String,
    @SerializedName("processedAt")   val processedAt: String?
)
